nuxeo-routing-core/src/main/java/org/nuxeo/ecm/platform/routing/core/impl/DocumentRoutingServiceImpl.java

```diff
@@ -1034,6 +1038,41 @@ public DocumentModelList runQuery() {
         }
     }
 
+    /**
+     * Cancel the workflow instance if all its attached document don't exist anymore. If the workflow is cancelled then
+     * the isWowkflowCanceled is set to true.
+     *
+     * @since 8.4
+     */
+    class AttachedDocumentsChecker extends UnrestrictedSessionRunner {
+
+        String workflowInstanceId;
+
+        Boolean isWowkflowCanceled = false;
```

Comment:
Well, shouldn't it be `boolean` then? I don't think the null case is used so a primitive is enough.